purldb-toolkit/tests/test_purlcli_mock.py

```diff
@@ -0,0 +1,1430 @@
+                            "pkg:gem/bundler-sass",
+                            "pkg:pypi/fetchcode",
+                            "pkg:pypi/fetchcode@0.1.0",
+                            "pkg:pypi/fetchcode@0.2.0",
+                        ],
+                        "tool_name": "purlcli",
+                        "tool_version": "0.1.0",
+                        "warnings": [
+                            "input PURL: 'pkg:pypi/fetchcode@0.1.0' normalized to 'pkg:pypi/fetchcode'",
+                            "input PURL: 'pkg:pypi/fetchcode@0.2.0' normalized to 'pkg:pypi/fetchcode'",
+                            "'pkg:gem/bundler-sass' not supported with `metadata` command",
+                        ],
+                    }
+                ],
+            ),
+        ],
+    )
+    def test_construct_headers_unique(self, test_input, expected):
+        metadata_headers = purlcli.construct_headers(
+            test_input,
+            output="",
+            file="",
+            command_name="metadata",
+            head=None,
+            normalized_purls=[
+                ("pkg:gem/bundler-sass", "pkg:gem/bundler-sass"),
+                ("pkg:pypi/fetchcode", "pkg:pypi/fetchcode"),
+                ("pkg:pypi/fetchcode@0.1.0", "pkg:pypi/fetchcode"),
+                ("pkg:pypi/fetchcode@0.2.0", "pkg:pypi/fetchcode"),
+            ],
+            unique=True,
+            purl_warnings={"pkg:gem/bundler-sass": "valid_but_not_supported"},
+        )
+        cli_test_utils.streamline_headers(expected)
+        cli_test_utils.streamline_headers(metadata_headers)
+
+        assert metadata_headers == expected
+
+
+class TestPURLCLI_urls(object):
+    @mock.patch("purldb_toolkit.purlcli.make_head_request")
+    def test_urls_cli_head(self, mock_make_head_request):
+        """
+        Test the `urls` command with actual and expected JSON output files.
+        """
+        mock_make_head_request.side_effect = [
+            {"get_request": "N/A"},
+            {"head_request": "N/A"},
+            {"get_request": "N/A"},
+            {"head_request": "N/A"},
+            {"get_request": "N/A"},
+            {"head_request": "N/A"},
+            {"get_request": 200},
+            {"head_request": 200},
+            {"get_request": 200},
+            {"head_request": 200},
+            {"get_request": 200},
+            {"head_request": 200},
+        ]
+        expected_result_file = test_env.get_test_loc(
+            "purlcli/expected_urls_output_head_mock.json"
+        )
+        actual_result_file = test_env.get_temp_file("actual_urls_output_head_mock.json")
+        options = [
+            "--purl",
+            "pkg:pypi/fetchcode",
+            "--head",
+            "--output",
+            actual_result_file,
+        ]
+        runner = CliRunner()
+        result = runner.invoke(purlcli.get_urls, options, catch_exceptions=False)
+        assert result.exit_code == 0
+
+        f_output = open(actual_result_file)
```

Comment:
Use a context manager when opening files: https://datagy.io/python-with-open/
```suggestion
        with open(actual_result_file) as f_output:
            output_data = json.load(f_output)
            cli_test_utils.streamline_headers(output_data["headers"])
```

Reply:
Context managers added for each file-open action in both test files.